src/palindrom.js

```diff
@@ -435,15 +435,18 @@ const Palindrom = (() => {
         }
 
         getPatchUsingHTTP(href) {
-            return this.xhr(
-                href,
-                'application/json-patch+json',
-                null,
-                (res, method) => {
-                    this.onReceive(res.data, href, method);
-                },
-                true
-            );
+            return new Promise(resolve => {
+                this.xhr(
+                    href,
+                    'application/json-patch+json',
+                    null,
+                    (res, method) => {
+                        this.onReceive(res.data, href, method);
+                        resolve();
+                    },
+                    true
+                );
+            });
```

Comment:
Shouldn't this promise reject on error?

BTW, maybe we could switch to [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)

Reply:
We have an old issue for this. https://github.com/Palindrom/Palindrom/issues/175

Reply:
Ok, WDYT about rejecting?

Reply:
I agree with @tomalec, it's a good practice with Promises is to always provide a meaningful rejection. In this case, `getPatchUsingHTTP` needs to:

-  `resolve` when the server responds with a correct patch (even if the response code is 4xx, see https://github.com/Palindrom/Palindrom/pull/221)
- `reject` when the server responds with something that is not a patch

Reply:
Yes I agree too! I'm working on it